File: PR1/PR1_Ej1/src/Bicicleta.java
public abstract class Bicicleta implements Runnable{
    protected int id = 0;
    protected String tipo = "normal";

    public abstract void mostrarTipo();

    public void setId(int id){
        this.id=id;
    }

    @Override
    public synchronized void run(){

    }
}

File: PR1/PR1_Ej2/Bicicleta.py
from abc import ABC, abstractmethod
from Protoype import Prototype
import copy

class Bicicleta(Prototype,ABC):
    def __init__(self):
        self.tipo="normal"

    @abstractmethod
    def mostrarTipo(self):
        pass

    @abstractmethod
    def clone(self):
        pass